src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/HashProviderDispenser.Windows.cs

```diff
@@ -208,6 +216,16 @@ public static unsafe int HashData(string hashAlgorithmId, ReadOnlySpan<byte> sou
                         Interop.BCrypt.BCryptAlgPseudoHandle.BCRYPT_SHA3_512_ALG_HANDLE;
                     digestSizeInBytes = SHA3_512.HashSizeInBytes;
                 }
+                else if (hashAlgorithmId == HashAlgorithmNames.CSHAKE128)
+                {
+                    algHandle = Interop.BCrypt.BCryptAlgPseudoHandle.BCRYPT_CSHAKE128_ALG_HANDLE;
+                    digestSizeInBytes = null;
+                }
+                else if (hashAlgorithmId == HashAlgorithmNames.CSHAKE256)
+                {
+                    algHandle = Interop.BCrypt.BCryptAlgPseudoHandle.BCRYPT_CSHAKE256_ALG_HANDLE;
+                    digestSizeInBytes = null;
```

Comment:
Why make digestSizeInBytes nullable, set these to null, and then later check if it's null, rather than just doing `digestSizeInBytes = destination.Length` in these two places?  The nullability just makes things more confusing, IMHO.

Reply:
I promise it was a very good idea when I started off, then I refactored my change in to something simpler, but didn't refactor it enough, and now it looks dumb.

Will fix.